server/src/building/ProjectBuildDatabse.cpp

```diff
@@ -0,0 +1,235 @@
+#include "ProjectBuildDatabase.h"
+#include "utils/GrpcUtils.h"
+#include "exceptions/CompilationDatabaseException.h"
+#include "utils/JsonUtils.h"
+#include "loguru.h"
+#include "utils/StringUtils.h"
+
+
+static std::string tryConvertOptionToPath(const std::string &possibleFilePath,
+                                          const fs::path &dirPath) {
+    if (StringUtils::startsWith(possibleFilePath, "-")) {
+        return possibleFilePath;
+    }
+    fs::path fullFilePath;
+    try {
+        fullFilePath = Paths::getCCJsonFileFullPath(possibleFilePath, dirPath);
+    } catch (...) {
+        return possibleFilePath;
+    }
+    return fs::exists(fullFilePath) ? fullFilePath.string() : possibleFilePath;
+}
+
+ProjectBuildDatabase::ProjectBuildDatabase(fs::path _buildCommandsJsonPath,
+                                           fs::path _serverBuildDir,
+                                           utbot::ProjectContext _projectContext) :
+        BuildDatabase(_serverBuildDir,
+                      _buildCommandsJsonPath,
+                      fs::canonical(_buildCommandsJsonPath / "link_commands.json"),
+                      fs::canonical(_buildCommandsJsonPath / "compile_commands.json"),
+                      std::move(_projectContext)) {
+    if (!fs::exists(linkCommandsJsonPath) || !fs::exists(compileCommandsJsonPath)) {
+        throw CompilationDatabaseException("Couldn't open link_commands.json or compile_commands.json files");
+    }
+
+    auto linkCommandsJson = JsonUtils::getJsonFromFile(linkCommandsJsonPath);
+    auto compileCommandsJson = JsonUtils::getJsonFromFile(compileCommandsJsonPath);
+
+    initObjects(compileCommandsJson);
+    initInfo(linkCommandsJson);
+    filterInstalledFiles();
+    addLocalSharedLibraries();
+    fillTargetInfoParents();
+    createClangCompileCommandsJson();
+}
+
+std::shared_ptr<ProjectBuildDatabase> ProjectBuildDatabase::create(const utbot::ProjectContext &projectContext) {
+    fs::path compileCommandsJsonPath =
+            CompilationUtils::substituteRemotePathToCompileCommandsJsonPath(
+                    projectContext.projectPath, projectContext.buildDirRelativePath);
+    fs::path serverBuildDir = Paths::getUtbotBuildDir(projectContext);
+    std::shared_ptr<ProjectBuildDatabase> buildDatabase = std::make_shared<ProjectBuildDatabase>(
+            std::move(ProjectBuildDatabase(compileCommandsJsonPath, serverBuildDir, projectContext)));
```

Comment:
`std::make_shared<ProjectBuildDatabase>(compileCommandsJsonPath, serverBuildDir, projectContext);` should work here, see [documentation](https://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared)